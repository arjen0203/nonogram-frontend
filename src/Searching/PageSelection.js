import React, {Component} from 'react';

class PageSelection extends Component {
    render() {
        return (
            <div data-testid="pageSelect" className="page-nav">
                <button className="page-nav-button" onClick={() => this.props.changePage(0)}>{"<<"}</button>
                <button className="page-nav-button" onClick={() => this.props.changePage(this.props.currentPage - 1)}>{"<"}</button>
                <label>
                <input data-testid="pageInput" className="page-nav-input" type="number" value={this.props.displayNumber} onChange={(e) => this.props.changeDisplayNumber(e.target.value)} onBlur={(e) => this.props.changePage(e.target.value)}></input>
                </label>
                <button className="page-nav-button" onClick={() => this.props.changePage(this.props.currentPage + 1)}>{">"}</button>
                <button className="page-nav-button" onClick={() => this.props.changePage(this.props.maxPage)}>{">>"}</button>
            </div>
        );
    }
}

export default PageSelection;