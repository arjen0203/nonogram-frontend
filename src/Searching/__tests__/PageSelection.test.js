import React from "react";
import ReactDOM from 'react-dom';
import PageSelection from "../PageSelection";
import {render, cleanup} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect"

import renderer from "react-test-renderer";

afterEach(cleanup);

it('Renders without crashing', function () {
	const div = document.createElement('div');
	ReactDOM.render(<PageSelection></PageSelection>, div);
});

it('Render correctly', function () {
	const mockPage = 1;

	const comp = render(<PageSelection displayNumber={mockPage}></PageSelection>).getByTestId("pageInput");
	expect(comp).toHaveValue(1);
});

it('Render correctly', function () {
	const mockPage = 6;

	const comp = render(<PageSelection displayNumber={mockPage}></PageSelection>).getByTestId("pageInput");
	expect(comp).toHaveValue(6);
});

it('matches snapshot', function () {
	const mockPage = 1;

	const tree = renderer.create(<PageSelection displayNumber={mockPage}></PageSelection>).toJSON();
	expect(tree).toMatchSnapshot();
});