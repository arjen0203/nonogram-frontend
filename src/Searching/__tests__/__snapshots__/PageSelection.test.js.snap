// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="page-nav"
  data-testid="pageSelect"
>
  <button
    className="page-nav-button"
    onClick={[Function]}
  >
    &lt;&lt;
  </button>
  <button
    className="page-nav-button"
    onClick={[Function]}
  >
    &lt;
  </button>
  <label>
    <input
      className="page-nav-input"
      data-testid="pageInput"
      onBlur={[Function]}
      onChange={[Function]}
      type="number"
      value={1}
    />
  </label>
  <button
    className="page-nav-button"
    onClick={[Function]}
  >
    &gt;
  </button>
  <button
    className="page-nav-button"
    onClick={[Function]}
  >
    &gt;&gt;
  </button>
</div>
`;
